### PR TITLE
F 703 support multi recipient

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
related to Giveth/giveth-dapps-v2#703

### CODE REVIEW

**Review by CarlosQ96 — APPROVED** [review]
Ignore this hahah

**Comment by mohammadranjbarz** [comment]
@MoeNick  @CarlosQ96 @MohammadPCh 

After merging this PR ( we are waiting for current release after that we would review and merge this)
The frontend should change two some web services:
### **createProject** and **updateProject**
Instead of sending `walletAddress` you should send this in the body of request

```
 addresses: [
        {
          address: walletAddress,
          networkId: 1 (for mainnet) or 3( for ropsten) or 100 (for xDai),
        },
      ]
```
### Getting projects
In every service that frontend gets project list or project detail , it would get address infoes as well with adding this to graphql query

```
 addresses {
          address
          isRecipient
          networkId
        }
```
And then based on each network that user selected on metamask frontend should use the address of that network, and make sure that the `isRecipient` is `true` for that address

**Comment by mohammadranjbarz** [comment]
@CarlosQ96 Please review it, but I keep it draft to not merge it until the frontend guys @MohammadPCh  or @MoeNick  tell us to merge it, because this PR changes are not backward compatible and staging may break if we merge it without implementing frontend side

**Review by CarlosQ96 — CHANGES_REQUESTED** [review]
https://github.com/Giveth/impact-graph/pull/525#issuecomment-1141686937 I think you mentioned the projectByAddress mutation is no longer used. We should delete it as it would be incomplete with this Pr, it would need to search by related addresses too.

Same for project query. As we do not user repository to fetch project.

File: src/resolvers/projectResolver.ts
Comment (CarlosQ96) [inline src/resolvers/projectResolver.ts:1106]: Sql injection can be done here, escape the address on the raw query.

File: src/repositories/projectAddressRepository.ts
Comment (CarlosQ96) [inline src/repositories/projectAddressRepository.ts]: We have an sql injection problem here. Specially since the query ```walletAddressIsPurpleListed``` accepts the address directly. We have to escape it!!

**Review by CarlosQ96 — CHANGES_REQUESTED** [review]

File: src/repositories/projectAddressRepository.ts
Comment (CarlosQ96) [reply src/repositories/projectAddressRepository.ts]: Did not find a clean way with raw queries... but maybe this can help.

https://github.com/typeorm/typeorm/issues/556#issuecomment-556042689

**Comment by CarlosQ96** [comment]
Quick question, is the previous purple address list lost? gotta reintroduce them in the adminbro?

**Comment by mohammadranjbarz** [comment]
> Quick question, is the previous purple address list lost? gotta reintroduce them in the adminbro?

Yeah we would lose them, but now there isn't any purple addresses in our DB and I told Ashley to not add any addresses until we merge this, so it would not make a problem

**Comment by CarlosQ96** [comment]
I think we can skip deleting the other mutations for now. The sql injection is fixed.

File: src/resolvers/projectResolver.ts
Comment (CarlosQ96) [inline src/resolvers/projectResolver.ts]: We have a N+1 queries issue here as we are saving entities separately. It's better to unify the jsons and insert once into the entity.

**Review by CarlosQ96 — CHANGES_REQUESTED** [review]

File: src/resolvers/projectResolver.ts
Comment (mohammadranjbarz) [reply src/resolvers/projectResolver.ts]: @CarlosQ96 Thanks for mentioning that, I fixed it here
https://github.com/Giveth/impact-graph/pull/531/commits/6fc6f71cac105fa28293b135418a3b9e1ee63517

**Review by CarlosQ96 — APPROVED** [review]
Awesome job man! All issues fixed. I think we only have to coordinate with @mateodaza to merge. He is wanting to integrate.

File: src/repositories/projectAddressRepository.ts
Comment (CarlosQ96) [reply src/repositories/projectAddressRepository.ts]: Fixed.!

Comment (mohammadranjbarz) [reply src/repositories/projectAddressRepository.ts]: Thanks so much

**Pull request merged**
